tests/index.test.js

```diff
@@ -0,0 +1,147 @@
+            text: "Hey! Don’t forget to bring the £5 you owe me. We’ll grab a coffee ☕ soon.",
+            expected: { encoding: 'UCS-2', numberOfSMS: 2, totalLength: 73, maxCharCount: 67 }
+        },
+        {
+            name: "French with special characters (UCS-2)",
+            text: "Salut! Comment ça va? J’espère que tout va bien avec toi et ta famille. On se voit ce week-end?",
+            expected: { encoding: 'UCS-2', numberOfSMS: 2, totalLength: 95, maxCharCount: 67 }
+        },
+        {
+            name: "German message (GSM-7)",
+            text: "Hallo! Ich hoffe, es geht dir gut. Wir sollten uns bald wieder treffen und über alte Zeiten sprechen.",
+            expected: { encoding: 'GSM-7', numberOfSMS: 1, totalLength: 101, maxCharCount: 160 }
+        },
+        {
+            name: "Spanish with special characters (UCS-2)",
+            text: "¡Hola! ¿Cómo estás? Tengo muchas ganas de verte pronto. No olvides traer los documentos.",
+            expected: { encoding: 'UCS-2', numberOfSMS: 2, totalLength: 88, maxCharCount: 67 }
+        },
+        {
+            name: "English with Euro symbol (GSM-7)",
+            text: "The price is €100, and we need to deliver it by Monday at the latest. Don't forget!",
+            expected: { encoding: 'GSM-7', numberOfSMS: 1, totalLength: 84, maxCharCount: 160 }
+        },
+        {
+            name: "English with Euro symbol (GSM-7)",
+            text: "The total cost is €50. We'll need to confirm the payment by tomorrow morning.",
+            expected: { encoding: 'GSM-7', numberOfSMS: 1, totalLength: 78, maxCharCount: 160 }
+        },
+        {
+            name: "Exact 160 character message (GSM-7)",
+            text: "This message is designed to be exactly 160 characters long. No more, no less. Let's make sure every word counts in this message and that it stays within the limit.",
+            expected: { encoding: 'GSM-7', numberOfSMS: 2, totalLength: 163, maxCharCount: 153 }
+        },
+        {
+            name: "Long English message exceeding 160 characters (GSM-7)",
+            text: "This is a long message that will exceed the 160 character limit for a single SMS message. When the length of this message exceeds the limit, it will be split into two or more parts.",
+            expected: { encoding: 'GSM-7', numberOfSMS: 2, totalLength: 181, maxCharCount: 153 }
+        },
+        {
+            name: "English message with a date and time (GSM-7)",
+            text: "Hello, John! Please confirm the date & time for our meeting at 10:00 AM. Call me at 555-1234. Thanks!",
+            expected: { encoding: 'GSM-7', numberOfSMS: 1, totalLength: 101, maxCharCount: 160 }
+        },
+        {
+            name: "GSM-7 line break test",
+            text: "Hello,\nThis is a message\nwith multiple line breaks.\nHow cool is that?\n\nBest regards,\nJohn",
+            expected: { encoding: 'GSM-7', numberOfSMS: 1, totalLength: 101, maxCharCount: 160 }
```

Comment:
Where is the `101` coming from here? Cause if I do a char count I get 89:

<img width="589" alt="Screenshot 2024-10-02 at 12 13 58" src="https://github.com/user-attachments/assets/4f958878-7287-45de-bbc5-7c3bd0b2c72f">

Reply:
![alt text](image.png) In front of every \n, an ESC character is included

Reply:
![image](https://github.com/user-attachments/assets/0d6b05bb-7ee8-491c-a867-41e82536b4b9)

Reply:
![image](https://github.com/user-attachments/assets/767e7101-754f-46bd-8f35-d5280ad6ca17)
Also in JS it counts `\n` as 1 - in SMS it's counted as 2

Reply:
6 line breaks = 12 extra chars in SMS (101-89=12)